zulipterminal/model.py

```diff
@@ -516,6 +516,59 @@ def send_stream_message(self, stream: str, topic: str, content: str) -> bool:
             notify_if_message_sent_outside_narrow(composition, self.controller)
         return message_was_sent
 
+    def is_unauthorised_to_post_in_stream(self, stream_id: int) -> Optional[str]:
+        """
+        Identify whether a user is authorised to send message to the given
+        stream or not using stream_post_policy and/or is_announcement_policy
+        and accordingly update the footer
+        """
+        stream_dict = self.stream_dict[stream_id]
+        user_info = self._all_users_by_id.get(self.user_id, None)
+        if not user_info:
+            # Check whether the given user id is valid or not
+            return "User does not exist"
+        if (
+            stream_dict.get("is_announcement_only")
+            or stream_dict.get("stream_post_policy") == 2
+        ):
+            # For admins and owners only (not using user role)
```

Comment:
In future servers there will likely be only `stream_post_policy`, so maybe we should check for the role here?

Could we simplify the conditionals overall, since `== 2` and `is_announcement_only` are the same?

Reply:
Actually, `role` falls under ZFL 59, whereas `is_admin`, `is_guest` and `is_owner` are under ZFL 8/10. Hence, I felt that we could do the checking without using `role`. However, I'll add `role`.

`is_announcement_only` got deprecated after [ZFL 1](https://zulip.com/api/update-stream#parameter-stream_post_policy) right? Hence, I decided to keep both so that it can function the same even after we stop support for v2.1.